tests/spicy/types/unit/print-hook.spicy

```diff
@@ -0,0 +1,52 @@
+# @TEST-EXEC:  ${SCRIPTS}/printf '\x00\x00\x00\x01\x02' | spicy-driver %INPUT >output
+# @TEST-EXEC:  btest-diff output
+
+module Mini;
+
+import spicy;
+
+public type Test = unit {
+    on %init   { print self; }
+    f1: uint32;
+    f2: uint8;
+    on %done   { print self; }
+
+    on %print  {
+        local f1 = "-";
+        local f2 = "-";
+
+        if ( self?.f1 )
+            f1 = "%s" % self.f1;
+
+        if ( self?.f2 )
+            f2 = "%s" % self.f2;
+
+        return "%s|%s" % (f1, f2);
+        }
+};
+
+@TEST-START-NEXT
+
+module Mini;
+
+import spicy;
+
+public type Test = unit {
+    on %init   { print self; }
+    f1: uint32;
+    f2: uint8;
+    on %done   { print self; }
+};
+
+on Test::%print {
+    local f1 = "-";
+    local f2 = "-";
+
+    if ( self?.f1 )
+        f1 = "%s" % self.f1;
+
+    if ( self?.f2 )
+        f2 = "%s" % self.f2;
+
+    return "%s|%s" % (f1, f2);
+}
```

Comment:
Can we add another test case for nested units? Given two units `A`, `B` and an `A` containing a `B`, test that if only `A`, only `B`, or both customizing their string repr, we get the expected result for an `A` defining
```.r
on A::%print {
  return "b: %s" % b;
}
```

Reply:
Good point, will do. Not quite sure actually what would be best for a subunit has %print with a parent that does not. Right now, that will use the sub's %print when the field is rendered inside the parent, which I think is ok, but could also see if differently (which would be harder though.)

Reply:
Naively I would also expect that if a subunit has a custom `%print` if would be use everywhere (e.g., also when the unit is wrapped; might want to add another test that this it also is used when `print` is invoked from outside any unit scope e.g., from a free, non-hook function).

The only issue I could see with that behavior is that there is no way to "remove" that print hook again from the outside to get back to the default behavior. But that is an issue of hooks in general which is why they generally shouldn't be part of grammars intended for reuse.

Reply:
Added that tests as well and hopefully fixed CI with a missing baseline update.